libs/cart/testing/src/drivers/in-memory/cart.service.spec.ts

```diff
@@ -57,46 +61,59 @@ describe('Driver | In Memory | Cart | CartService', () => {
     beforeEach(() => {
       productId = 'productId';
       qty = 1;
-
-      mockCart = cartFactory.create();
     });
 
     describe('a successful addToCart request', () => {
-      it('should send a post request to `api/cart/addToCart` and respond with a cart', () => {
+      it('should send a post request to `api/cart/addToCart` and respond with a cart', done => {
         cartService.addToCart(productId, qty).subscribe(cart => {
           expect(cart).toEqual(mockCart);
-        });  
-  
+          done();
+        });
+
         const req = httpMock.expectOne(`${cartService.url}/addToCart`);
-  
+
         expect(req.request.method).toBe('POST');
         expect(req.request.body).toEqual({
-          'productId': productId,
-          'qty': qty
+          productId,
+          qty
         });
-        
+
         req.flush(mockCart);
       });
     });
   });
 
   describe('clear', () => {
-
-    beforeEach(() => {
-      mockCart = cartFactory.create();
-    });
-
     describe('a successful clear request', () => {
-      it('should send a post request to `api/cart/clear` and not return a value', () => {
-        cartService.clear().subscribe();
-        
+      it('should send a post request to `api/cart/clear` and not return a value', done => {
+        cartService.clear(cartId).subscribe(res => {
+          expect(res).not.toBeTruthy();
+          done();
+        });
+
         const req = httpMock.expectOne(`${cartService.url}/clear`);
 
         expect(req.request.method).toBe('POST');
-        expect(req.request.body).toEqual({});
+        expect(req.request.body).toEqual({cartId});
 
-        req.flush(mockCart);
+        req.flush(null);
       });
     });
   });
+
+  describe('create | creating a cart', () => {
+    it('should send a post request to `api/cart/create` and return the cart', done => {
+      cartService.create().subscribe(result => {
+        expect(result).toEqual(mockCart);
```

Comment:
This expectation should be failing based upon the interface described.

Reply:
Fixed the message.

Reply:
No, I think that the test should actually be failing as described. `create` does not return a `mockcart`.

Reply:
Changed the expected and flushed value to just an object with ID.